Lake/Util/Compare.lean

```diff
@@ -47,13 +47,34 @@ theorem Nat.eq_of_compare
 {n n' : Nat} : compare n n' = Ordering.eq → n = n' := by
   simp only [compare]; exact eq_of_compareOfLessAndEq
 
+@[simp]
+theorem Nat.compare_iff_eq
+{n n' : Nat} : compare n n' = Ordering.eq ↔ n = n' := by
```

Comment:
BTW, you might want to consider upgrading the `EqOfCmp` class to `EqIffCmp` (with the `∀ a b, compare a b = .eq ↔ a = b` property).  This gives you a very nice simp lemma and make the proofs a lot smoother.

Reply:
This is true. Note that `eq_iff_cmp` is a strictly stronger requirement than `eq_of_cmp`, though, as it also requires that the compare be reflexive (i.e., that `compare a a = .eq`), which is not necessarily true. However, since `EqOfCmp` is already more general than the class directly required by `DRBMap` (i.e., `EqOfCmpWrt`), generalizing it some more likely couldn't hurt. It also should probably be called `CmpIffEq` based on the ordering in the theorems.